src/components/AlumniTimeline.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import styles from './AlumniTimeline.module.css';
+import Members from "./Members";
+
+const AlumniOfClass = ({classOf, alumni}) => {
+  return (
+    <li className={styles.event}>
+      <label className={styles.icon}></label>
+      <div className={styles.body}>
+        <p className={styles.date}>Class of {classOf}</p>
+          <Members members={alumni} />
+      </div>
+    </li>
+  )
+}
+
+const AlumniTimeline = ({alumni}) => {
+  // JSX elements for all alumni
+  const listItems = [];
+  // All alumni of a certain graduating class
+  let alumniOfClass = [];
+  let lastClassOf = null;
+
+  alumni.forEach((alumnus) => {
+    if (alumnus.classOf !== lastClassOf && lastClassOf !== null) {
+      // Adding JSX Elements for class of `lastClassOf`
+      listItems.push(
+        <AlumniOfClass
+          classOf={lastClassOf}
+          alumni={alumniOfClass}
+          key={lastClassOf} />
+      );
+      // Emptying the alumniOfClass list to accept new ones
+      alumniOfClass = [];
+    }
+    alumniOfClass.push(alumnus);
+    lastClassOf = alumnus.classOf;
+  });
```

Comment:
The alumni list partitioning step should happen outside of the component. Then the component can use `.map` to render without an imperative `forEach`

Reply:
Thought so, but the `alumni` array is received through `props` and is restricted in the functional scope.

Or perhaps you mean declaring a function to do the partitioning job?

Reply:
Instead of receiving the flattened `alumni` array, you could choose to receive a partitioned one.
For example
```tsx
// members.tsx
type AlumnusFromYear = { readonly year: number; readonly members: readonly Member[] };
const partition = (alumnus: readonly Member[]): readonly AlumnusFromYear[] => { ... };

<AlumniTimeline partitionedAlumniList={partition(alumnus)} />

// AlumniTimeline.tsx
return (
  <div>
    {partitionedAlumniList.map(({ year, members }) => (
      <div>
        {year}
        {members.map(...)}
    ))}
  </div>
)
```

Reply:
Yeah! Got it